### DISCUSSION TITLE
Fields input types

### DISCUSSION THREAD

**Discussion by ElTacitos** [johnlindquist/kit#1137] [Q&A] [answered]
Hey,
Is there a place where I can find all the types a field input can have ? I have found that it can be `text` and `number`.
I am trying to figure out if I can add a choice array to a fields input ^^

**Comment by johnlindquist** [comment, marked as answer]
@ElTacitos No, there's no choices for fields inputs. The alternatives are `arg` (the most common) or `template` or `form`:

```js
let cars = ["Volvo", "Saab", "Mercedes", "Audi"]

let text = await template(
  `This is my car: \${1|${cars.join(",")}|}`
)

let result = await form(`
<label for="cars">Choose a car:</label>

<select name="cars" id="cars">
    ${cars
      .map(car => {
        return `<option value="${car}">${car}</option>`
      })
      .join("")}
</select>
`)

await dev({
  text,
  result,
})
```

**Reply by ElTacitos** [reply]
Thanks a lot I'll look into it 👌
